shogun-lib/src/main/java/de/terrestris/shogun/lib/service/security/permission/GroupInstancePermissionService.java

```diff
@@ -111,26 +112,61 @@ public void setPermission(BaseEntity persistedEntity, Group group, PermissionCol
             throw new RuntimeException("Could not find requested permission collection");
         }
 
-        Optional<GroupInstancePermission> existingPermissions = findFor(persistedEntity, group);
+        clearExistingPermission(group, permissionCollection.get(), persistedEntity);
+
+        GroupInstancePermission groupInstancePermission = new GroupInstancePermission();
+        groupInstancePermission.setGroup(group);
+        groupInstancePermission.setEntityId(persistedEntity.getId());
+        groupInstancePermission.setPermissions(permissionCollection.get());
+
+        repository.save(groupInstancePermission);
+    }
+
+    /**
+     * Set Permission for SHOGun group for multiple entities at once
+     * @param persistedEntityList A collection of entities to set permission for
+     * @param group The SHOGun group
+     * @param permissionCollectionType The permission collection type (e.g. READ, READ_WRITE)
+     */
+    public void setPermissionBulk(
```

Comment:
Non blocking, but in terms of naming harmonization I would suggest to rename the method to `setPermission`.

Reply:
fixed